src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/EnclaveAzureDatabaseTests.cs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.using System;
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider;
+using Microsoft.Data.SqlClient.ManualTesting.Tests;
+using Xunit;
+using Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.Setup;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted
+{
+    public class EnclaveAzureDatabaseTests : IDisposable
+    {
+        private ColumnMasterKey akvColumnMasterKey;
+        private ColumnEncryptionKey akvColumnEncryptionKey;
+        private SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider; 
+        private List<DbObject> databaseObjects = new List<DbObject>();
+        private List<string> connStrings = new List<string>();
+             
+        public EnclaveAzureDatabaseTests()
+        {
+            if (DataTestUtility.IsEnclaveAzureDatabaseSetup() && DataTestUtility.EnclaveEnabled)
+            {
+                // Initialize AKV provider
+                sqlColumnEncryptionAzureKeyVaultProvider = new SqlColumnEncryptionAzureKeyVaultProvider(AADUtility.AzureActiveDirectoryAuthenticationCallback);
+
+                // Register AKV provider
+                SqlConnection.RegisterColumnEncryptionKeyStoreProviders(customProviders: new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(capacity: 1, comparer: StringComparer.OrdinalIgnoreCase)
+                {
+                    { SqlColumnEncryptionAzureKeyVaultProvider.ProviderName, sqlColumnEncryptionAzureKeyVaultProvider}
+                });
+
+                akvColumnMasterKey = new AkvColumnMasterKey(DatabaseHelper.GenerateUniqueName("AKVCMK"), akvUrl: DataTestUtility.AKVUrl, sqlColumnEncryptionAzureKeyVaultProvider, DataTestUtility.EnclaveEnabled);
+                databaseObjects.Add(akvColumnMasterKey);
+
+                akvColumnEncryptionKey= new ColumnEncryptionKey(DatabaseHelper.GenerateUniqueName("AKVCEK"),
+                                                              akvColumnMasterKey,
+                                                              sqlColumnEncryptionAzureKeyVaultProvider);
+                databaseObjects.Add(akvColumnEncryptionKey);
+
+                SqlConnectionStringBuilder connString1 = new SqlConnectionStringBuilder(DataTestUtility.EnclaveAzureDatabaseConnString);
+                connString1.InitialCatalog = "testdb001";
```

Comment:
This test class is very specific to our DB setup, so also add that in comments above as "For Internal Use only"